### DISCUSSION TITLE
ubuntu gpsd connection

### DISCUSSION THREAD

**Discussion by FreaktheBear** [Stellarium/stellarium#3936] [Q&A]
Dear All,
First a big shout out to the developers for all the hard work!
Just finished installing version 24.3 on a Raspberry Pi 4B 8GB running Ubuntu 24.04.1 LTS.
I used the instructions at the bottom of the Stellarium Raspberry Pi Wiki page.
I can start Stellarium with the command QT_QPA_PLATFORM=xcb ./stellarium --opengl-compat in the folder /usr/bin.
Still need to somehow add this variable to the /usr/share/org.stellarium.Stellarium.desktop, but that is for later concern.

I have a U-blox NEO-7 GPS connected via USB: https://www.jaycar.co.nz/arduino-compatible-gps-receiver-module/p/XC3710 
I installed the packages gpsd, gpsd-clients, ntp and configured the /etc/default/gpsd file with:
START_DAEMON="true"
DEVICES="ttyACM0"
GPSD_OPTIONS="-G"
USBAUTO="true"

The GPS works, showing a 3D fix in the "cgps -s" application (part of the gpsd-clients package).
Also "xgps" shows the connection with localhost:2947 and a fix.
However Stellarium is not able to connect to the GPS and shows "GPS Failed".
The Log in the Help (F1) menu shows:
Cannot open serial port to NMEA device at port "ttyACM0"

Question: Why does Stellarium trying to open the physical port? Shouldn't it connect to the gpsd daemon?

**Comment by 10110111** [comment]
Do you see a message like the following in the log?

```
LibGPSHelper not ready. Attempting a direct NMEA connection instead.
```
If not, this may mean that you built Stellarium without libgps enabled. See the output of CMake for details.

**Reply by FreaktheBear** [reply]
Hi Ruslan,
Thank you for your prompt reply!
I cannot see the log message which you are referring to.
I did not build Stellarium from the source files, but used the package installer.
I followed the Stellarium Raspberry Pi Wiki stating the following:

 2024-05-12: On Ubuntu 24.4 arm64, install raspi-config and activate wayland. Then install Stellarium from ppa:stellarium/daily, and call with: QT_QPA_PLATFORM=xcb ./stellarium --opengl-compat. You may set the environment variable permanently and configure another, STEL_OPTS=--opengl-compat, for a permanent setting.

Kind regards

**Reply by gzotti** [reply]
Have you seen the GPSD-specific documentation in the Stellarium User Guide? I have developed this a few years ago with a borrowed device and have written these instructions so I don't forget what to do.

**Reply by 10110111** [reply]
@alex-w Do the builds on Launchpad install `libgps-dev` before running `cmake`? From the log of a failed arm64 build I see that it didn't (not sure how to get the log from a successful build).

**Reply by alex-w** [reply]
> @alex-w Do the builds on Launchpad install `libgps-dev` before running `cmake`? From the log of a failed arm64 build I see that it didn't (not sure how to get the log from a successful build).

No, see branches qt6 and qt64 - https://github.com/Stellarium/stellarium-ppa/branches - I'll add GPS support

**Reply by FreaktheBear** [reply]
Dear Team,

Thank you so much for the prompt actions!
I will await the qt6/qt64 build with GPS support.
I will also have an extra look at the manual remarks about "/lib/udev/60-gpsd.rules" which under Ubuntu 24.04 resides under "/lib/udev/rules.d/60-gpsd.rules".
I did use the "-G" option to start gpsd which I found in the manual.

Kind regards and good night from this side of the planet :-)

**Reply by alex-w** [reply]
https://code.launchpad.net/~stellarium/+recipe/stellarium-release-qt6.4

**Reply by gzotti** [reply]
I was not aware of non-inclusion of libgps-dev. Wait for the new package (or build yourself, takes just about 20 minutes on a (cooled) RPI4) before playing with the settings, of course.

**Reply by FreaktheBear** [reply]
Dear Team,

Thank you for the quick response and solution!
I updated Stellarium from the repository and the "Get location from GPS" worked straight away and shows "3D location fix".

Kind regards
